src/blockdata/transaction.rs

```diff
@@ -452,6 +452,24 @@ impl Transaction {
         Ok(())
     }
 
+    #[cfg(feature="bitcoinconsensus")]
+    /// Verify that this transaction is able to spend its inputs
+    /// with provided script interpreter flags. Flags are applied
+    /// globally for all transaction inputs.
+    /// The lambda spent should not return the same TxOut twice!
```

Comment:
Hmm I recall the conversations on another proposal to add a `verify` method on `Transaction` that had this requirement. I'm not sure how I feel about this requirement.

I think personally I would prefer a message like "this method doesn't check for calls to the `spent` function with the same key twice, so it is up to the user to check for duplicate inputs".

At least that makes it possible to pass `inputHashMap.get` or some generic closure and do the duplicate check before.